app/controllers/UserTokenController.scala

```diff
@@ -113,7 +114,9 @@ class UserTokenController @Inject()(datasetDAO: DatasetDAO,
 
     def tryRead: Fox[UserAccessAnswer] =
       for {
-        dataSourceBox <- datasetDAO.findOneByNameAndOrganization(dataSourceId.name, dataSourceId.team).futureBox
+        dataSourceBox <- datasetDAO
+          .findOneByDirectoryNameAndOrganization(dataSourceId.directoryName, dataSourceId.organizationId)
```

Comment:
maybe a shortcut called findOneByDataSourceId could be created in the DAO, which does this expansion and forwards to findOneByDirectoryNameAndOrganization